core/arch/arm/plat-imx/link.mk

```diff
@@ -2,10 +2,15 @@ include core/arch/arm/kernel/link.mk
 
 all: $(link-out-dir)/tee-raw.bin
 
+cleanfiles += $(link-out-dir)/entry_point_address.txt
+$(link-out-dir)/entry_point_address.txt: $(link-out-dir)/tee.elf
+	@$(cmd-echo-silent) '  GEN     $@'
+	$(q)$(READELFcore) -h $(link-out-dir)/tee.elf | awk '/Entry point address/ {print $$4}' > $@
+
 .PHONY: uTee
 uTee: $(link-out-dir)/uTee
 cleanfiles += $(link-out-dir)/uTee
-$(link-out-dir)/uTee: $(link-out-dir)/tee-raw.bin
+$(link-out-dir)/uTee: $(link-out-dir)/tee-raw.bin $(link-out-dir)/entry_point_address.txt
 	@$(cmd-echo-silent) '  MKIMAGE $@'
-	$(q)ADDR=`printf 0x%x $$(($(CFG_TZDRAM_START)))`; \
+	$(q)@ADDR=`cat $(link-out-dir)/entry_point_address.txt`; \
```

Comment:
`$(subst UL,,$(CFG_TZDRAM_START))` would work too...

Reply:
That would work indeed. I thought about doing something similar but since there are other similar attributes like `U`, `ULL`, `L` ... We would also have to consider the other cases ... no?

Reply:
Why would a physical address ever need to be encoded as an unsigned long long? IMO it should always be a unsigned long, and often an unsigned (int) is enough and we drop the UL. But that's it, no?

Reply:
Knowing that `CFG_TZDRAM_START` could be the result of multiple additions, I thought that considering all these attributes was also a good idea.
But yes, we should only get `UL`.